### ISSUE TITLE
uper constraints not validated

### ISSUE THREAD

**Issue by shahn** [librasn/rasn#286]
Hi, and thanks for rasn! I am wondering if my expectations are wrong or if there is a missing feature/bug. I am using rasn-compiler to generate a struct. I would have expected that the call to `encode()` would return an error, but too large values are accepted. In addition, depending on how much too large the value is, a roundtrip might still succeed with `decode()` also not checking the constraint.

```
#[derive(AsnType, Debug, Clone, Decode, Encode, PartialEq, PartialOrd, Eq, Ord, Hash)]
#[rasn(delegate, value("-500..=504"))]
pub struct Test(pub i16);

#[test]
fn it_works() {
    let x = Test(505);
    let output = uper::encode(&x).unwrap(); // I would have expected this to fail
    let y: Test = uper::decode(&output).unwrap();
    assert_eq!(x, y); // This assert does not trigger

    let x = Test(815);
    let output = uper::encode(&x).unwrap();
    let y: Test = uper::decode(&output).unwrap();
    assert_eq!(x, y); // This assert triggers with 815 != -209
}
```

In addition, it is possible to trigger a debug_assert inside rasn with this code:
```
    let x = Test(-501);
    let output = uper::encode(&x).unwrap();
```

It fails here:
```
panicked at src/types/constraints.rs:432:13:
assertion failed: value >= start
```

Am I somehow fundamentally misunderstanding the intended API, or can I help in adding the functionality if it is missing currently?

**Comment by shahn** [comment]
I tried to add the validation for Integers, maybe that's a start? I think it is sufficient for my usecase for now

**Comment by Nicceboy** [comment]
Hey, I checked most versions back to the version when UPER was added, and there seems to be no check. Encoding has never raised an error if it is out of range; I guess constraint is just used to encode it "correctly". However, it is not always correct if the value is outside of the range too much. Decoder tries to then decode based on correct range, and there is a mismatch. OER currently raises error when value is out of range, so I think it is okay add for PER too.
